Homework_10_Messenger

ira-01 opened the pull request:
Development of a software  messenger

ira-01 committed:
Homework_10/Messenger

Development of a software  messenger

rodikno commented on Homework_10_InMessenger/InMessenger/src/Main.java:
```diff
@@ -0,0 +1,61 @@
+        users.put("Diana", new User("Diana"));
+
+        // Adding contacts
+        users.get("Kate").addContact(users.get("Martin"));
```
Розумію, що ти намагалась робити усі маніпуляції з юзерами, коли вони вже є членами колекції, але це трохи ускладнює читання/розуміння клієнтського коду, хоч і зекономило трохи місця.

`User` - референсний тип, тому все таки можна було ініціалізувати цих трьох окремо
```java
User kate = new User("Kate");
User martin = new User("Martin");
User diana = new User("Diana");
```

нижче зробити просто масив з референсами на них, потім гратися з ними як з окремими об*єктами (відправляти повідомлення один одному), а внизу з масиву вже перебиратись по ним щоб виводити їх повідомлення і т.п. - так було б чистіше, бо погодься подібні штуки трохи важкувато читаються, хоч і працюють:

```java
users.get("Kate").sendMessage(users.get("Martin"), "Hello, Martin!");
```

rodikno commented on Homework_10_InMessenger/InMessenger/src/Main.java:
```diff
@@ -0,0 +1,61 @@
+        // Reading messages
+        System.out.println("\nMartin reads his messages:");
+        for (Message message : users.get("Martin").getMessages()) {
+            users.get("Martin").readMessage(message);
```
Виглядає як кандидат на додавання цієї логіки юзеру, в деякий метод `public void readAllMessages()`

rodikno commented on Homework_10_InMessenger/InMessenger/src/Main.java:
```diff
@@ -0,0 +1,61 @@
+
+        // Delete contact and message
+        users.get("Martin").removeContact(users.get("Diana"));
+        System.out.println("\nMartin removed Diana from contacts.");
```
Виведення цього меседжу можна сховати в сам метод `.removeContact()`, так клієнтський код буде чистішим, а виведення логічно пов*язане з тим, що відбувається)

rodikno commented on Homework_10_InMessenger/InMessenger/src/Main.java:
```diff
@@ -0,0 +1,61 @@
+
+        // Checking, if the list is not empty before deleting the message
+        if (!sentMessages.isEmpty()) {
+            users.get("Martin").removeMessage(sentMessages.get(0));
```
Тут не дуже зрозуміло що відбувається: ти намагаєшся видалити нульовий відправлений меседж?

Якщо мета - очистити власні відправлені (як в телеграмі "очистити чат тільки для себе") - тоді це можна винести в метод класу User, і тоді вже перебиратися по колекції власних меседжів і видаляти усі зі статусом `SENT`

rodikno commented on Homework_10_InMessenger/InMessenger/src/User.java:
```diff
@@ -0,0 +1,74 @@
+    public void readMessage(Message message) {
+        if (messages.contains(message) && message.getStatus() == MessageStatus.RECEIVED) {
+            message.setStatus(MessageStatus.READ);
+        }
```
`else` ?

rodikno reviewed (changes requested):
Привіт! Загалом непогано, але можна трохи допрацювати, щоб зробити класи ефективнішими, логіку простішою, а клієнтський код - чистішим і зрозумілішим)

Допушувати можна в туж бранчу, потім просто пересабміть пул реквест і відміть домашку виконаною в гугл класрум пліз, щоб я побачив

ira-01 committed:
Improvements made

The code has been corrected; the logic is simpler, and the client code is cleaner and more understandable.
